libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/list/DirectionListAdapter.java

```diff
@@ -74,7 +74,7 @@ public void clear() {
   }
 
   private void addLegSteps(RouteProgress routeProgress) {
-    if (newLeg(routeProgress) && legHasSteps(routeProgress)) {
+    if ((newLeg(routeProgress) || legSteps.size() == 0) && legHasSteps(routeProgress)) {
```

Comment:
intentional? were you trying to correct another issue?

Reply:
Yeah, we weren't repopulating the directions list during re-route scenarios (with the new legs) - the size will be zero because clear the list when we get an off-route